samples/matter/common/src/bridge/ble_connectivity_manager.cpp

```diff
@@ -163,6 +163,11 @@ void BLEConnectivityManager::DiscoveryCompletedHandler(bt_gatt_dm *dm, void *con
 			provider->GetBLEBridgedDevice().mFirstConnectionCallback(
 				dm, discoveryResult, provider->GetBLEBridgedDevice().mFirstConnectionCallbackContext);
 			provider->ConfirmInitialConnection();
+		} else {
+			/* Provider has been already initialized and we need to parse the received data to attach the
+			 * gatt handle after Bridge reboot */
+			VerifyOrReturn(provider->ParseDiscoveredData(dm) == 0,
```

Comment:
Besides loosing the GATT data in case of a reconnect, the subscriptions also get lost. 
Subscribe is only called in the providers Init function (not yet for onoff, but e.g. for environmental data provider).
So would this be the correct place to also call provider->Init()?

Reply:
Actually, `provider->Init()` will be called in the other place, to be more precise, here: https://github.com/nrfconnect/sdk-nrf/blob/main/samples/matter/common/src/bridge/bridge_manager.cpp#L287 so calling it in `DiscoveryCompletedHandler` is redundant. A different situation is with `ParseDiscoveredData` because this method is called within the `mFirstConnectionCallback` callback, but when the device is restoring from the memory (after the Bridge reboot) it is not called because we have a provider assigned as `InitiallyConnected`.

Reply:
You are right. I can see that there is a call stack from AppTask::RestoreBridgedDevices() to the given location. In that case, Subscribe() will be called before getting the log "app: Connected: 62:19:96:A4:E0:DF (random)" resulting in "Invalid connection object". (means before being connected)
The second call stack to reach the given location is via BluetoothDeviceConnected(...) which is only reached from the shell command add.

You mentioned that ParseDiscoveredData will not be called when restoring. Actually ParseDiscoveredData is required to fill in the mGattPostureSubscribeParams. Without these, Subscription will also fail and the polling fallback is also missing the correct mPostureCharacteristicHandle.

Actually the the issue with the Subscribe is not really related to this merge request. Just thought this is an opportunity for feedback as there is no "Issues" tab on this repository.

Reply:
Hmm, ok you're right! :) We've not noticed that and focused on the LBS sample, but yes, a subscription should be called after the connection establishment, not in the init() method. Thank you for the input.